### ISSUE TITLE
[java] ImmutableField should consider enum class

### ISSUE THREAD

**Issue by Jane-PL-256** [pmd/pmd#3957]
**Affects PMD Version:6.42.0** 

**Rule:ImmutableField**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield>

**Description:**
HI, I found a false negative about the rule `ImmutableField` and it is caused by enum class. Please read the following code example. PMD should have reported a warning in line 4 because `f` is a immutable field. Hence, I think this is a false negative caused by anonymous class. Thanks for your consideration.

**Code Sample demonstrating the issue:**
```java
public class D {
    enum E {
        R;
        private int y = 42;  // should have a warning in this line
    }
}
```
But the following code example can be detected:
```java
public class D {
    private int x = 42;  // can report a warning in this line
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
